ui/components/User.js

```diff
@@ -167,81 +156,61 @@ class User extends React.Component {
-              aria-owns={open
-                ? 'menu-list-grow'
-                : undefined}
-              aria-haspopup="true"
-              onClick={this.handleToggle}
             >
-              <Avatar className={avatarClassName} src={avatar_url} imgProps={{ referrerPolicy : "no-referrer" }} />
-            </IconButton>
-          </div>
-          <Popper open={open} anchorEl={this.anchorEl} transition style={{ zIndex : 10000 }} placement="top-end">
-            {({ TransitionProps, placement }) => (
-              <Grow
-                {...TransitionProps}
-                id="menu-list-grow"
-                style={{
-                  transformOrigin : placement === 'bottom'
-                    ? 'left top'
-                    : 'left bottom'
-                }}
-              >
-                <Paper className={classes.popover}>
-                  <ClickAwayListener onClickAway={this.handleClose}>
-
-                    <MenuList>
-
-                      {
-                        this.state.account && this.state.account.length ?
-                          (
-                            <>
-                              {this.renderAccountExtension(this.state.account)}
-                            </>
-                          )
-                          :
-                          null
-                      }
-                      <MenuItem onClick={this.handleGetToken}>Get Token</MenuItem>
-                      <MenuItem onClick={this.handlePreference}>Preferences</MenuItem>
-                      <MenuItem onClick={this.handleLogout}>Logout</MenuItem>
-                    </MenuList>
-                  </ClickAwayListener>
-                </Paper>
-              </Grow>
-            )}
-          </Popper>
-        </NoSsr>
-      </div>
-    );
-  }
-}
+              <Paper className={classes.popover}>
+                <ClickAwayListener onClickAway={handleClose}>
+                  <MenuList>
+                    {account && account.length ? (
+                      <>
+                        {renderAccountExtension(account)}
+                      </>
+                    ) : null}
+                    <MenuItem onClick={handleGetToken}>Get Token</MenuItem>
+                    <MenuItem onClick={handlePreference}>Preferences</MenuItem>
+                    <MenuItem onClick={handleLogout}>Logout</MenuItem>
+                  </MenuList>
+                </ClickAwayListener>
+              </Paper>
+            </Grow>
+          )}
+        </Popper>
+      </NoSsr>
+    </div>
+  );
 
-const mapDispatchToProps = (dispatch) => ({ updateUser : bindActionCreators(updateUser, dispatch), });
-const mapStateToProps = state => ({
-  capabilitiesRegistry : state.get("capabilitiesRegistry")
-})
+}
 
-export default withStyles(styles)(connect(
-  mapStateToProps,
-  mapDispatchToProps,
-)(withNotify((withRouter(User)))));
\ No newline at end of file
+export default withStyles(styles)((withNotify((withRouter(User)))));
```

Comment:
We need to remove the `withNotify ` and `withRouter` from here. Instead, we need to use `useNotification` and `useRouter` .

`useNotification` is a custom hook and `useRouter` is a pre-defined hook in next.js.